tests/storages_tests/test_with_server.py

```diff
@@ -211,3 +235,88 @@ def test_get_best_trial(direction: StudyDirection, values: Sequence[float]) -> N
+
+
+def test_set_and_get_study_user_attrs_for_floats() -> None:
+    storage = get_storage()
+    study_id = storage.create_new_study(directions=[StudyDirection.MINIMIZE])
+
+    # Test setting value.
+    for key, value in FLOAT_ATTRS.items():
+        storage.set_study_user_attr(study_id, key, value)
+        assert is_equal_floats(storage.get_study_user_attrs(study_id)[key], value)
+
+
+def test_set_and_get_study_system_attrs_for_floats() -> None:
+    storage = get_storage()
+    study_id = storage.create_new_study(directions=[StudyDirection.MINIMIZE])
+
+    # Test setting value.
+    for key, value in FLOAT_ATTRS.items():
+        storage.set_study_system_attr(study_id, key, value)
+        assert is_equal_floats(storage.get_study_system_attrs(study_id)[key], value)
+
+
+def test_set_trial_state_values_for_floats() -> None:
+    storage = get_storage()
+    study_id = storage.create_new_study(directions=[StudyDirection.MINIMIZE])
+    for value in FLOAT_ATTRS.values():
+        if math.isnan(value):  # NOTE: Optuna does not accept `nan` as `value`.
+            continue
+        trial_id = storage.create_new_trial(study_id)
+        storage.set_trial_state_values(trial_id, state=TrialState.COMPLETE, values=(value,))
+        set_value = storage.get_trial(trial_id).value
+        assert set_value is not None
+        assert is_equal_floats(set_value, value)
+
+
+def test_set_trial_param_for_floats() -> None:
+    storage = get_storage()
+    study_id = storage.create_new_study(directions=[StudyDirection.MINIMIZE])
+    trial_id = storage.create_new_trial(study_id)
+
+    for key, value in FLOAT_ATTRS.items():
+        float_distribution = FloatDistribution(low=value, high=value)
+        categorical_distribution = CategoricalDistribution(choices=(value,))
+        for distribution in (float_distribution, categorical_distribution):
+            # NOTE: suggest_float does not generate NaN, and MySQL cannot handle infinities.
+            if isinstance(distribution, FloatDistribution) and not math.isfinite(value):
+                continue
+            param_name = distribution.__class__.__name__ + key
+            internal_repr = distribution.to_internal_repr(value)
+            storage.set_trial_param(trial_id, param_name, internal_repr, distribution)
+            assert is_equal_floats(storage.get_trial_param(trial_id, param_name), internal_repr)
+            assert storage.get_trial(trial_id).distributions[param_name] == distribution
+
+
+def test_set_trial_intermediate_value_for_floats() -> None:
+    storage = get_storage()
+    study_id = storage.create_new_study(directions=[StudyDirection.MINIMIZE])
+    trial_id = storage.create_new_trial(study_id)
+    for i, value in enumerate(FLOAT_ATTRS.values()):
+        storage.set_trial_intermediate_value(trial_id, i, value)
+        assert is_equal_floats(storage.get_trial(trial_id).intermediate_values[i], value)
+
+
+def test_set_trial_user_attr_for_floats() -> None:
```

Comment:
Same as I mentioned in `test_storages.py`.
```suggestion
def test_set_and_get_trial_user_attr_for_floats() -> None:
```